internal/certificate/data_source_test.go

```diff
@@ -65,7 +66,7 @@ func TestAccHcloudDataSourceCertificateTest(t *testing.T) {
 	})
 }
 
-func TestAccHcloudDataSourceCertificateListTest(t *testing.T) {
+func TestAccCertificateDataSource_List(t *testing.T) {
```

Comment:
We have 2 different "kind" of datasources, single and list. I'd suggest to use: `DataSource` for single and `DataSourceList` for lists. And then if we have specific tests, we can add the extra suffix `_Managed` or `_Uploaded`

 ```suggestion
func TestAccCertificateDataSourceList(t *testing.T) {
```